packages/csharp/ArmoniK.Api.Mock/CountingService.cs

```diff
@@ -0,0 +1,156 @@
+                                                               .Any())
+                        .Select(type => type!)
+                        .ToList();
+
+    Counts = Services
+      // Create a dictionary for all the types to record method calling counts
+      .ToDictionary(type => type!.Name,
+                    type => type!.GetMethods()
+                                 // Get all methods that have the [Count] attribute
+                                 .Where(method => method.GetCustomAttributes<CountAttribute>()
+                                                        .Any())
+                                 .ToDictionary(method => method.Name,
+                                               _ => new StrongBox<long>(0)));
+  }
+
+  /// <summary>
+  ///   Get a list of all counting services (ie: types that are marked with [Counting] attribute)
+  /// </summary>
+  /// <returns>Enumeration of counting services</returns>
+  public static IEnumerable<Type> GetServices()
+    => Services;
+
+  /// <summary>
+  ///   Get the counters for all recorded types and methods.
+  /// </summary>
+  /// <param name="exclude">Methods to exclude from the count</param>
+  /// <returns>Counters</returns>
+  public static Dictionary<string, Dictionary<string, long>> GetCounters(IDictionary<string, HashSet<string>>? exclude = null)
+    => Counts.ToDictionary(kvType => kvType.Key,
+                           kvType =>
+                           {
+                             var              typeName      = kvType.Key;
+                             HashSet<string>? methodExclude = null;
+                             exclude?.TryGetValue(typeName,
+                                                  out methodExclude);
+                             methodExclude ??= new HashSet<string>();
+
+                             return kvType.Value.Where(kv => !methodExclude.Contains(kv.Key))
+                                          .ToDictionary(kv => kv.Key,
+                                                        kv => kv.Value.Value);
+                           });
+
+  /// <summary>
+  ///   Reset all the counters
+  /// </summary>
+  public static void ResetCounters()
+  {
+    foreach (var counts in Counts)
+    {
+      foreach (var count in counts.Value)
+      {
+        count.Value.Value = 0;
+      }
+    }
+  }
+}
+
+/// <summary>
+///   Mark a class as being a counting service
+/// </summary>
+[AttributeUsage(AttributeTargets.Class)]
+[UsedImplicitly]
+public class CountingAttribute : Attribute
+{
+}
+
+/// <summary>
+///   Mark a method to count the number of calls made to this method.
+/// </summary>
+[AttributeUsage(AttributeTargets.Method)]
+[UsedImplicitly]
+public class CountAttribute : Attribute, IMethodDecorator
+{
+  public void Init(object     instance,
+                   MethodBase method,
+                   object[]   args)
+  {
+    _ = args;
```

Comment:
```suggestion
                   object[]   _)
  {
```

Reply:
I think I prefer `_ = args;` as it does not redefine the name `_`.